animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,649 @@
+---
+layout: page
+title: "Shared Element Transitions"
+permalink: /animations/shared-element-transitions/
+---
+
+{% comment %}
+XXX - TODO
+
+* My tweaked copy of basic_radial_transition doesn't look right.
+  It takes on an oval shape as it flies.
+* Currently the adb command isn't working. I IM'd on Hackers
+  and received the following from Amir Hardon:
+  "Hi Shams (: a shot in the dark for the adb
+  error - I think I had similar issues when the command line adb
+  is from a different Android sdk than the one used by Android
+  studio (I think Android studio keep trying to restart adb when
+  you kill it), you can easily check by closing Android studio,
+  killing the adb server and then trying again
+  (while studio is closed)"
+  "If that's the issue a better fix would be to make sure the
+  SDK you use from the command line is the same one intellij is
+  using."
+  Solution: Hans made the videos for me.
+* Eventually publish these under GoogleDevelopers on Youtube but,
+  in the meantime, publish them on my Youtube account.
+{% endcomment %}
+
+<div class="whats-the-point" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What you'll learn:</b>
+
+* Create a shared element transition using Flutter's Hero widget.
+* Fly an element from one page to another.
+* Animate morphing a circular shape to a rectangular shape.
+</div>
+
+This guide demonstrates how to build shared element transitions
+in Flutter, also called shared element _animations_.
+There are several types of shared element transitions, and this
+guide covers two of them.
+
+* TOC Placeholder
+{:toc}
+
+<aside class="alert alert-info" markdown="1">
+**New to Flutter?**
+You should first understand how to create a layout using Flutter’s
+widgets.  For more information, see [Building Layouts in
+Flutter](/tutorials/layout/).
+</aside>
+
+A shared element transition animates an element (typically an image)
+from one page to another. This guide focuses on:
+
+<ul>
+<li markdown="1">Flying transitions<br>
+A _flying transition_ animates the element
+from one page to a new page, usually landing at a different
+location and with a different size.
+
+The following video (recorded at slow speed) shows a typical example.
+Clicking the flippers in the center of the screen flies them to the
+upper left corner of a new, blue screen, at a smaller size.
+Clicking the flippers in the blue screen
+(or using the device's back-to-previous-screen gesture)
+returns the user to the original screen.
+
+<center><iframe width="560" height="315" src="https://www.youtube.com/embed/uF7_JfhBsww" frameborder="0" allowfullscreen></iframe></center>
```

Comment:
Hope we can find a better solution than YouTube. It's really not ideal. In addition to showing a wide player for a vertical video, it suggests irrelevant videos after the demo video finishes. It can be very distracting. See an example below:

![image](https://user-images.githubusercontent.com/348942/31424168-a4b378c6-ae0d-11e7-8fe9-2ceecb91e235.png)

Reply:
I spoke to Seth about this last week, @InMatrix. He said that if we can find a place to host the content, we can present a more customized solution (in portrait mode, for example), but it's more work to set up. For now, I'm taking the easy route but, I agree, we need a better solution.